qa/rpc-tests/sporks.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Dash Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.mininode import *
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from time import *
+
+'''
+'''
+
+class SporkTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 3
+        self.setup_clean_chain = True
+        self.is_network_split = False
+
+    def setup_network(self):
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir,
+                                     ["-debug", "-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"]))
+        self.nodes.append(start_node(1, self.options.tmpdir,
+                                     ["-debug"]))
+        self.nodes.append(start_node(2, self.options.tmpdir,
+                                     ["-debug"]))
+        # connect only 2 first nodes at start
+        connect_nodes(self.nodes[0], 1)
+
+    def get_test_spork_state(self, node):
+        info = node.spork('show')
+        # use InstantSend spork for tests
+        return info['SPORK_2_INSTANTSEND_ENABLED'] < time()
+
+    def set_test_spork_state(self, node, state):
+        if state:
+            value = 0
+        else:
+            value = 4070908800
+        # use InstantSend spork for tests
+        node.spork('SPORK_2_INSTANTSEND_ENABLED', value)
+
+    def run_test(self):
+        # check test spork default state
+        assert(self.get_test_spork_state(self.nodes[0]))
+        assert(self.get_test_spork_state(self.nodes[1]))
+        assert(self.get_test_spork_state(self.nodes[2]))
+
+        # check spork propagation for connected nodes
+        self.set_test_spork_state(self.nodes[0], False)
+        start = time()
+        sent = False
+        while True:
```

Comment:
pls add a small timeout between rpc calls in `while True` loops e.g. `sleep(0.1)`